x/gamm/types/msgs.go

```diff
@@ -215,9 +215,10 @@ func (msg MsgJoinSwapExternAmountIn) ValidateBasic() error {
 	if err != nil {
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "Invalid sender address (%s)", err)
 	}
-
-	if !msg.TokenIn.IsValid() || !msg.TokenIn.IsPositive() {
-		return sdkerrors.Wrap(sdkerrors.ErrInvalidCoins, msg.TokenIn.String())
+	for _, coin := range msg.TokensIn {
```

Comment:
Would `msg.TokensIn.IsAllPositive` work here?

Reply:
No, IsAllPositive is sdk.coins' method. In order to be able to use it we first need to do something like this: 
```
tokensIn := sdk.NewCoins(msg.TokensIn...)
```
However, sdk.NewCoins method validates coins passed:
```
if err := newCoins.Validate(); err != nil {
	panic(fmt.Errorf("invalid coin set %s: %w", newCoins, err))
}
```
And as you can see it panics, not returns an error. Hence, test cases with intentionally wrong denoms panic, but we do not implement a catch for these situations currently

Reply:
Isn't `msg.TokensIn` already a `Coins` object? If so, why not just call `Validate`?

Reply:
@alexanderbez this is what I thought when first doing that, however, msg.TokensIn is []sdk.Coin, which makes .IsValid not a method for this type
<img width="906" alt="Screen Shot 2022-10-19 at 11 26 30" src="https://user-images.githubusercontent.com/46343690/196597840-a274e872-b67d-456c-a508-330e7eddda27.png">